ldrb/ldrb.py

```diff
@@ -357,18 +358,35 @@ def apex_to_base(
 
     base_bc = df.DirichletBC(V, 1, ffun, base_marker, "topological")
 
-    def solve(solver_parameters):
-        df.solve(a == L, apex, base_bc, solver_parameters=solver_parameters)
-
-    solver_parameters = {"linear_solver": "cg", "preconditioner": "amg"}
-    pcs = (pc for pc in ["petsc_amg", "default"])
-    while 1:
-        try:
-            solve(solver_parameters)
-        except RuntimeError:
-            solver_parameters["preconditioner"] = next(pcs)
-        else:
-            break
+    # Solver options
+    ksp_type = "cg"
+    ksp_norm_type = "unpreconditioned"
+    ksp_rtol = 1e-9
+    ksp_atol = 1e-15
+    ksp_max_it = 10000
+    ksp_error_if_not_converged = False
+    ksp_monitor = True
+    ksp_view = True
+    pc_type = "hypre"
+    pc_hypre_type = "boomeramg"
+    pc_view = True
```

Comment:
Some of these options, like `ksp_view`, `ksp_monitor` and `pc_view`, are only intended for debugging. They should probably be turned off by default.

Reply:
I added a verbose flag to turn these on and off (and they are by default off). If you set the log level to debug in the main function then the verbose flag will be on.